src/Mod/TechDraw/Gui/ViewProviderPage.cpp

```diff
@@ -195,10 +194,7 @@ bool ViewProviderPage::onDelete(const std::vector<std::string>&)
     // the ExportName of a template always begins with "Template"
     bool isTemplate = false;
     for (auto objsIterator : objs) {
-        if (objsIterator->getExportName().substr(0, 8).compare(std::string("Template")) == 0)
-            isTemplate = true;
-        else
-            isTemplate = false;
+        isTemplate = (objsIterator->getExportName().substr(0, 8).compare(std::string("Template")) == 0);

```

Comment:
Out of scope for this PR:

TODO: make a local property/function ilass for testing whether template.